snuba/datasets/events.py

```diff
@@ -200,6 +225,90 @@ def get_promoted_context_columns(self):
     def get_required_columns(self):
         return self.__required_columns
 
+    def column_expr(self, column_name, body):
+        if NESTED_COL_EXPR_RE.match(column_name):
+            return self._tag_expr(column_name)
+        elif column_name in ['tags_key', 'tags_value']:
+            return  self._tags_expr(column_name, body)
+        elif column_name == 'issue':
+            return  'group_id'
+        elif column_name == 'message':
+            # Because of the rename from message->search_message without backfill,
+            # records will have one or the other of these fields.
+            # TODO this can be removed once all data has search_message filled in.
+            return  'coalesce(search_message, message)'
+        else:
+            return  escape_col(column_name)
+
+
+    def _tag_expr(self, column_name):
+        """
+        Return an expression for the value of a single named tag.
+
+        For tags/contexts, we expand the expression depending on whether the tag is
+        "promoted" to a top level column, or whether we have to look in the tags map.
+        """
+        col, tag = NESTED_COL_EXPR_RE.match(column_name).group(1, 2)
+
+        # For promoted tags, return the column name.
+        if col in self.get_promoted_columns():
+            actual_tag = self.get_tag_column_map()[col].get(tag, tag)
+            if actual_tag in self.get_promoted_columns()[col]:
+                return string_col(self, actual_tag)
+
+        # For the rest, return an expression that looks it up in the nested tags.
+        return u'{col}.value[indexOf({col}.key, {tag})]'.format(**{
+            'col': col,
+            'tag': escape_literal(tag)
+        })
+
+
+    def _tags_expr(self, column_name, body):
+        """
+        Return an expression that array-joins on tags to produce an output with one
+        row per tag.
+        """
+        assert column_name in ['tags_key', 'tags_value']
+        col, k_or_v = column_name.split('_', 1)
+        nested_tags_only = state.get_config('nested_tags_only', 1)
+
+        # Generate parallel lists of keys and values to arrayJoin on
+        if nested_tags_only:
+            key_list = '{}.key'.format(col)
+            val_list = '{}.value'.format(col)
```

Comment:
Any chance of UnicodeError here?

Reply:
Hmm, not sure - but since this was already this way I don't want to change this functionally here